### ISSUE TITLE
Install `magic` tool instead of `modular`?

### ISSUE THREAD

**Issue by chriselrod** [Sharktheone/arch-mojo#27]
https://github.com/modularml/mojo/issues/3459#issuecomment-2349280107

`modular` has been deprecated in favor of `magic`.

**Comment by Sharktheone** [comment]
Ah, thanks, I need to see what I can do to replace with magic.

**Comment by Sharktheone** [comment]
It looks like you can just use the normal installation instructions from modular to install `magic`. I'll still be looking to add an AUR package for it.

**Comment by xentropic-dev** [comment]
Thanks, I needed this.  I was getting a ton of errors running "mojo test" regarding libncurses, and started manually trying to fix them myself.  Then I found you already had this nice script and it works perfectly with the new magic installer.  Thank you!

**Comment by incansvl** [comment]
Modular (the company) seem to have switched a lot of their documentation on how to get started with mojo over to `magic`. It's difficult to follow those on an install based on `modular` (the command) as the tools are more the same.

**Comment by Sharktheone** [comment]
> Modular (the company) seem to have switched a lot of their documentation on how to get started with mojo over to `magic`. It's difficult to follow those on an install based on `modular` (the command) as the tools are more the same.

Yes, you should use the `magic` tool to install mojo now. The installation command in the docs should also work on arch.

To the general `magic` situation with this repo:
I have a PKGBUILD for magic, but when I asked modular if they would be okay with me packaging magic, I got some responses from the community that they might not like it, but I haven't got any response from Modular.

In case you get into issues about missing libraries, you need to install the `mojo-libs` package from the AUR.

(I should probably update the readme...)

**Comment by incansvl** [comment]
The installation of `magic` from the Modular web site only claims to support Ubuntu, and as far as i can see involves basically downloading and running a script with root/sudo privileges. I'm wary of that because I could envisage it breaking stuff, at least in terms of Ubuntu vs. Arch conventions for where things are installed (PATH entries etc.)

I was also under the impression that having an AUR-style package would also give the option for a clean uninstall as well (true?) whereas the good 'ol "download a script and run it" approach tends to be a one way trip, with no easy way to back out the installation if it causes problems.

Even if `magic` provides for clean installation / de-installation of mojo and the other things it manages (python packages etc.?) the question is, what provides for clean installation + de-installation of `magic` itself?  It's not clear that the community comments consider this.

**Comment by Sharktheone** [comment]
> The installation of `magic` from the Modular web site only claims to support Ubuntu, and as far as i can see involves basically downloading and running a script with root/sudo privileges. I'm wary of that because I could envisage it breaking stuff, at least in terms of Ubuntu vs. Arch conventions for where things are installed (PATH entries etc.)

If you don't want to use the script from Modular, you can use the `PKGBUILD` at https://github.com/Sharktheone/arch-mojo/blob/main/aur/magic/PKGBUILD (make sure to also copy the `magic.install` file)



> I was also under the impression that having an AUR-style package would also give the option for a clean uninstall as well (true?) whereas the good 'ol "download a script and run it" approach tends to be a one way trip, with no easy way to back out the installation if it causes problems.
> 
> Even if `magic` provides for clean installation / de-installation of mojo and the other things it manages (python packages etc.?) the question is, what provides for clean installation + de-installation of `magic` itself? It's not clear that the community comments consider this.

afaik magic only installs things in `~/.modular/` and the venvs your project directories. 
There is also `magic clean` which *should* delete all venvs if I understood that correctly

**Comment by incansvl** [comment]
I tried the PKGBUILD approach you suggested, and it seems to have worked (thanks), but I have a question.

I was surprised after running the install to find that the `magic` binary wasn't copied into place in a directory in the PATH (in this case `/home/username/.modular/bin/magic`) but is a symlink back to the directory that I ran the PKGBUILD installation from-

`204810 lrwxrwxrwx 1 username username 79 Mar 18 15:34 /home/username/.modular/bin/magic -> /home/username/Downloads/Linux/arch-mojo/aur/magic/magic-x86_64-unknown-linux-musl`

I'm new to Arch and not that familiar with the conventions. However in other environments i'm used to installation working folders being temporary (disposable). In this case if I go back and clean up my installation dirs, i'll delete the magic binary, probably without realising it.

Should the executable have been copied into place, so there is no dependency on the installation directory?

**Comment by Sharktheone** [comment]
> I was surprised after running the install to find that the `magic` binary wasn't copied into place in a directory in the PATH (in this case `/home/username/.modular/bin/magic`) but is a symlink back to the directory that I ran the PKGBUILD installation from-


This probably comes from the fact how the fake root system works in arch. So basically each PKGBUILD gets executed in a fake root environment where it can do all sorts of funny things without completely affecting the host. However when moving a file in the fake root it will just get symlinked on the host (at least that's how I understand it, no guarantee for correctness)

**Comment by incansvl** [comment]
Ok. I see that makes sense in terms of trying to prevent bad (through error or malice) installation packages wreaking haoc on a user's machine. I means I need to rethink my understanding of what the installation working directory is.
